### PR TITLE
Homework/gmock

### PR DESCRIPTION
**Pull request by kuchichan**
Ready for review

### CODE REVIEW

**Review by WBigaj — COMMENTED** [review]
I have checked your code and it looks ok.

File: test/PizzeriaTest.cpp [hunk 1]
```diff
@@ -65,3 +65,33 @@ TEST_F(PizzeriaTest, calculatePriceForPizzaMock)
...
 }
+
+TEST_F(PizzeriaTest, MainCppScenarioTest) {
+    // There is no need to expect_call for NiceMock 
+    StrictMock<PizzaMock> mockeritha{};
+    NiceMock<PizzaMock> mockeroni{};
+    
+    auto stubbioni = new PizzaStub{"Stubbioni"};
+
+    Pizzas pizzas = {&mockeritha, &mockeroni, stubbioni};
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:76]: great names :D

File: test/PizzeriaTest.cpp [hunk 2]
```diff
@@ -65,3 +65,33 @@ TEST_F(PizzeriaTest, calculatePriceForPizzaMock)
...
+    EXPECT_CALL(mockeritha, getBakingTime()).WillOnce(Return(minutes(10)));
+
+    EXPECT_CALL(mt, sleep_for(minutes(0))).Times(1);
+    EXPECT_CALL(mt, sleep_for(minutes(10))).Times(1);
+    EXPECT_CALL(mt, sleep_for(minutes(1))).Times(1);
+
+    pizzeria.bakePizzas(orderId);
+    pizzeria.completeOrder(orderId);
+
+    ASSERT_EQ(calculatedPrice, 57.50);
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:94]: please avoid magic numbers. Use constexpr for values like 22.50, 25.00, minutes(10), etc.
Then use them in proper places. It shows dependencies between objects.

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
{ "requirement_ids" : [1119, 1119, 1119, 3524, 3525, 3526] }

3 * 3 XP za punktualność (1119)
10 XP za TimeMock w Pizzas (3524)
15 XP za nowy przypadek testowy w Pizzas (3525)
5 XP za workflow w Pizzas (3526)

**Pull request closed without merging**